.eslintrc.js

```diff
@@ -1,45 +1,23 @@
 module.exports = {
-  env: {
-    browser: true,
-    es6: true,
-  },
   extends: [
-    'standard',
-    'standard-react',
-    'prettier/react',
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'prettier/@typescript-eslint',
     'plugin:prettier/recommended',
+    'react-app',
   ],
-  parser: 'babel-eslint',
+  plugins: ['@typescript-eslint', 'prettier'],
   parserOptions: {
-    ecmaFeatures: {
-      experimentalObjectRestSpread: true,
-      jsx: true,
-    },
-    sourceType: 'module',
-  },
-  plugins: ['prettier', 'react', 'react-hooks'],
-  rules: {
-    'react/prop-types': 'warn',
-    'react-hooks/rules-of-hooks': 'error',
-    'react-hooks/exhaustive-deps': 'error',
-    'import/no-unresolved': [
-      'error',
-      { ignore: ['^react(-dom)?$', '^styled-components$'] },
-    ],
-    'promise/no-nesting': ['off'],
-    'linebreak-style': ['error', 'unix'],
-    curly: 'error',
+    jsx: true,
+    project: './tsconfig.json',
   },
   settings: {
     react: {
-      pragma: 'React',
-      version: '16.6',
+      version: '999.999.999',
```

Comment:
What would you think about doing this?

```suggestion
      version: require('./package.json').peerDependencies.react,
```

Reply:
I think keeping the config free of side effects would be simper and more portable. For consistency I've reverted to being explicit with the version and pragma.